### DISCUSSION TITLE
Backup straight to QCOW2?

### DISCUSSION THREAD

**Discussion by billystaggs** [abbbi/virtnbdbackup#191] [General]
I am moving from ESXi to KVM and I'm attempting to workout a backup solution before migrating.  I've got virtnbdbackup working with full and inc backups, rotating directories nightly.  However I'm concerned with the time to restore a backup.  Running virtnbdrestore to restore a backup into a workable qcow2 file is time consuming and will no doubt increase my recovery time should I have an outage.  I have successfully used "virsh backup-begin" to backup straight to a qcow2, but the docs seem difficult for me to grasp and I don't like the external xml files. I would rather explorer the opportunity to do this with virtnbdbackup.  So is this possible? 

Thanks,
--
bstaggs

**Comment by abbbi** [comment]
See issue #57 
Could be implemented But currently i dont have the Time to do so.. it would also require some implementation to merge the inc/diff Backups back into a base Image

Copying qcow Files and merging them back to a usable base image takes some time, too

maybe current restore performance could be improved by certain options to the involved qemu tools (amount of threads Or using asyncio) But i have Not done any Performance tests:

https://github.com/abbbi/virtnbdbackup/blob/0e723baf634a63fd9faa462561b8aa947bc5544d/libvirtnbdbackup/qemu/util.py#L97

one could play with the qemu-nbd options here that might enhance throughput (such as --cache or --aio)
